src/proto/streams/streams.rs

```diff
@@ -1141,17 +1197,17 @@ impl Actions {
         buffer: &mut Buffer<Frame<B>>,
         stream: &mut store::Ptr,
         counts: &mut Counts,
-        res: Result<(), RecvError>,
+        res: RecvError,
```

Comment:
I suppose it was mostly the following lines:
https://github.com/carllerche/h2/blob/3197bd69a885ab3033caed925210c39f96cd4c85/src/proto/streams/streams.rs#L473-L479

Before the function transformed a stream reset "error" into the same case as no error at all, and I wanted to treat those differently.
Felt more ergonomic at that point to just handle the error case only in that function, since it's all the function looks at anyway